media/server/gstplayer/source/Utils.cpp

```diff
@@ -133,4 +136,65 @@ std::string getUnderflowSignalName(const firebolt::rialto::wrappers::IGlibWrappe
     }
     return "";
 }
+
+GstCaps *createCapsFromMediaSource(const std::shared_ptr<firebolt::rialto::wrappers::IGstWrapper> &gstWrapper,
```

Comment:
There is a lot of repeated logic here, maybe we could refactor the if-else logic into a function  to encapsulate the decision making for the createCapsFromMediaSource function. We could maybe introduce a function to the config types and then update the createCapsFromMediaSource to use that function - just to reduce duplication of code and making it easier to read

Reply:
it's an implementation moved from AttachSource. We can create a new task to refactor it, if you think, that it's needed :)

Reply:
Ahh yeh I see, I still think it would simplify the logic in createCapsFromMediaSource making it easier to read, and it would make it easier to add new media types (in the future)

Reply:
might also need to adjust the ut as well then